core/src/navigation/services/semi-collapsed-navigation.js

```diff
@@ -7,11 +7,20 @@ class SemiCollapsibleNavigationClass {
     this.responsiveNavSetting = LuigiConfig.getConfigValue(
       'settings.responsiveNavigation'
     );
-    this.semiCollapsible =
+    let currentCollapsedState = localStorage.getItem(
```

Comment:
My suggestion:
```
let currentCollapsedState = JSON.parse(localStorage.getItem(
      NavigationHelpers.COL_NAV_KEY
    ));
```
In `currentCollapsedState` you will get a string instead of a boolean value.

Reply:
done